tests/python_package_test/test_dask.py

```diff
@@ -1527,6 +1541,43 @@ def test_predict_with_raw_score(task, output, cluster):
             assert_eq(raw_predictions, pred_proba_raw)
 
 
+@pytest.mark.parametrize("output", data_output)
+@pytest.mark.parametrize("use_init_score", [False, True])
+def test_predict_stump(output, use_init_score, cluster, rng):
+    with Client(cluster) as client:
+        task = "binary-classification"
+        n_samples = 1_000
+        _, _, _, _, dX, dy, _, dg = _create_data(objective=task, n_samples=n_samples, output=output)
+
+        model_factory = task_to_dask_factory[task]
+
+        params = {"objective": "binary", "n_estimators": 5, "min_data_in_leaf": n_samples}
+
+        if not use_init_score:
+            init_scores = None
+        elif output.startswith("dataframe"):
+            init_scores = dy.map_partitions(lambda x: pd.DataFrame(rng.uniform(size=x.size)))
+        else:
+            init_scores = dy.map_blocks(lambda x: rng.uniform(size=x.size))
+
+        model = model_factory(client=client, **params)
+        model.fit(dX, dy, group=dg, init_score=init_scores)
+        preds_1 = model.predict(dX, raw_score=True, num_iteration=1).compute()
+        preds_all = model.predict(dX, raw_score=True).compute()
```

Comment:
```suggestion
        _, _, _, _, dX, dy, _, _ = _create_data(objective="binary-classification", n_samples=1_000, output=output)

        params = {"objective": "binary", "n_estimators": 5, "min_data_in_leaf": n_samples}

        if not use_init_score:
            init_scores = None
        elif output.startswith("dataframe"):
            init_scores = dy.map_partitions(lambda x: pd.DataFrame(rng.uniform(size=x.size)))
        else:
            init_scores = dy.map_blocks(lambda x: rng.uniform(size=x.size))

        model = lgb.DaskLGBMClassifier(client=client, **params)
        model.fit(dX, dy, group=dg, init_score=init_scores)
        preds_1 = model.predict(dX, raw_score=True, num_iteration=1).compute()
        preds_all = model.predict(dX, raw_score=True).compute()
```

I support hard-coding this to just binary classification, as I did in the `test_engine.py` test, in the interest of saving CI time.

If we're going to do that hard-coding though, let's take advantage of it to simplify the test code a bit.